src/main/java/com/researchspace/dataverse/http/DataverseOperationsImplV1.java

```diff
@@ -203,7 +193,46 @@ public List<DatasetVersion> getDatasetVersions (Identifier dsIdentifier) {
 		handleError(resp);
 		return resp.getBody().getData();
 	}
-	
+
+	@Override
+	public DatasetFileList uploadNativeFile( byte[] data, FileUploadMetadata metadata, Identifier dsIdentifier,  String fileName){
+		ByteArrayResource resource = new ByteArrayResource(data){
+			@Override
+			public String getFilename(){
+				return fileName;
+			}
+		};
+		return getDatasetFileList(metadata, dsIdentifier, resource);
+	}
+    // TODO stream read twice???
```

Comment:
What do you mean by read stream twice? Do you have any problem? We can discuss it if you want.